core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/shared/services/dot-edit-page-resolver/dot-edit-page-resolver.service.ts

```diff
@@ -25,37 +25,21 @@ export class DotEditPageResolver implements Resolve<DotPageRenderState> {
     constructor(
         private dotPageStateService: DotPageStateService,
         private dotRouterService: DotRouterService,
-        private dotHttpErrorManagerService: DotHttpErrorManagerService
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private siteService: SiteService
     ) {}
 
     resolve(route: ActivatedRouteSnapshot): Observable<DotPageRenderState> {
         const data = this.dotPageStateService.getInternalNavigationState();
+        const renderOptions = this.getDotPageRenderOptions(route);
+        const currentSection = route.children[0].url[0].path;
+        const isLayout = currentSection === 'layout';
+        const hostId = route.queryParams?.host_id;
```

Comment:
what happens if this is `null` ?
would it break on `line 98` ???

Reply:
If the `hostId` is equal to `null` or `undefined`, we don't switch the `site`. There is a test for that case: 

https://github.com/dotCMS/core/blob/b6d62b3757669ad289d11d67dcedefef5aacbc8c/core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/shared/services/dot-edit-page-resolver/dot-edit-page-resolver.service.spec.ts#L166-L176